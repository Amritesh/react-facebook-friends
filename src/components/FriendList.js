import React, { Component, PropTypes } from 'react';
import FriendItem from './FriendItem';
import Radium from 'radium';
import { MAX_OUTPUT } from '../utils/constants';

class FriendList extends Component {

  constructor(props) {
    super(props);
  }

  static propTypes = {
    myFriends: PropTypes.array.isRequired
  }

  renderFriends(nodata) {

    let { myFriends, query } = this.props;
    const len = myFriends.length > MAX_OUTPUT ? MAX_OUTPUT : myFriends.length;

    const result = myFriends.slice(0, len).reduce((prev, curr, i) => {

      if (curr.name.match(new RegExp(query, 'i'))) {
        prev.push(<FriendItem key={i} {...curr} rank={i + 1} />);
      }

      return prev;

    }, []);

    return result.length > 0 ? result : (<div style={nodata}>No results for: "{query}"</div>);
  }

  render() {

    const contentStyle = {
      padding: '2em 1em 0',
      '@media (min-width: 48em)': {
        padding: '2em 3em 0',
        marginLeft: '25%'
      },
      nodata: {
        color: 'white'
      }
    };

    return (
      <div className="pure-u-1 pure-u-md-3-4" style={contentStyle}>
        <div>
          {this.renderFriends(contentStyle.nodata)}
        </div>
      </div>
    );
  }
}

export default Radium(FriendList);
