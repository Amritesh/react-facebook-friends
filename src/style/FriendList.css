.nodata {
  font-size: 1.5em;
  display: flex;
  justify-content: center;
  align-items: center;
  text-align: center;
  color: white;
  min-height: 100vh;
}

.wrapper {
  flex: 3;
  display: flex;
  flex-direction: column;
  align-items: center;
}

@media (max-width: 1050px) {
  .wrapper {
    flex: 1 1 100%;
  }

  .nodata {
    min-height: auto;
  }
}