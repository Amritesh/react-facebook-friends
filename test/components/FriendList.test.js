import React from 'react';
import renderer from 'react-test-renderer';
import FriendList from '../../src/components/FriendList';

test('FriendList component should renders correctly', () => {
  const props = {
    myFriends: [{
      COMMENT: 3,
      LIKE: 1,
      POST: 2,
      id: '123',
      link: 'link',
      name: 'name',
      score: 30,
      url: 'url'
    }],
    query: 'q'
  };
  const component = renderer.create(<FriendList {...props} />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
