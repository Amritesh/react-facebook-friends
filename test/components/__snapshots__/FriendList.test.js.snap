// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FriendList component should renders correctly 1`] = `
<div
  className="wrapper"
>
  <div
    className="nodata"
  >
    No results for: "q"
  </div>
</div>
`;
